### ISSUE TITLE
About reproducing Emotion-LLaMA F1 score on MER2023

### ISSUE THREAD

**Issue by ArD2** [ZebangCheng/Emotion-LLaMA#93]
Hello author, I am trying to obtain the exact results as shown in your paper, when evaluating in the MER2023-SEMI; I have followed the process of training in the README.md. My results after the stage 2 were the following:
```
Accuracy: 0.8225419664268585
Precision: 0.8318784850552208
Recall: 0.8225419664268585
F1 Score: 0.8176724092912279
[[153   1   2  22   0   5]
 [  0 147  10   7   4   1]
 [  4  16 107  24   1  14]
 [  1   1   0 254   0   1]
 [  1   0   0   8   3   2]
 [  3   1   1  18   0  22]]
}
```

However this results were obtained using the MERR_coarse_grained.txt in the featureface.yaml. After looking through [#69 ](https://github.com/ZebangCheng/Emotion-LLaMA/issues/69) I have used the MER2023_pseudo_label.txt to train the stage 1 (in both cases I train stage 2 with MERR_fine_grained.txt) and now I have obtained the following results:
```
Accuracy: 0.8932853717026379
Precision: 0.8918073472309769
Recall: 0.8932853717026379
F1 Score: 0.891229884725899
[[172   1   6   2   0   2]
 [  0 154  10   3   1   1]
 [  7  11 135   5   1   7]
 [  1   4   1 248   0   3]
 [  1   0   1   3   5   4]
 [  4   1   3   6   0  31]]
```

Although there is an improvement, there is still a difference between 0.8912 and 0.9036. I would like to ask if there is an additional step to get into the same results. Also, worth to mention that in the dataset_file (firstface.py) I only used the "emotion" task.

Terminal when evaluating:
```
CUDA_VISIBLE_DEVICES=0 torchrun eval_emotion.py --cfg-path eval_configs/eval_emotion.yaml --dataset feature_face_caption

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please submit your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
================================================================================
cfg: Namespace(cfg_path='eval_configs/eval_emotion.yaml', name='A2', ckpt=None, eval_opt='all', max_new_tokens=10, batch_size=32, lora_r=64, lora_alpha=16, options=None, dataset=['feature_face_caption'], res=100.0, resample=False)
Initialization Model
Loading checkpoint shards: 100%|█████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.83it/s]
loraconfig: LoraConfig(peft_type=<PeftType.LORA: 'LORA'>, base_model_name_or_path=None, task_type='CAUSAL_LM', inference_mode=False, r=64, target_modules=['q_proj', 'v_proj'], lora_alpha=16, lora_dropout=0.05, merge_weights=False, fan_in_fan_out=False, enable_lora=None, bias='none', modules_to_save=None)
trainable params: 33554432 || all params: 6771970048 || trainable%: 0.49548996469513035
Position interpolate from 16x16 to 32x32
Load Minigpt-4-LLM Checkpoint: /home/arturo/Emotion-LLaMA-main/checkpoints/save_checkpoint/20250617160/checkpoint_29.pth
Initialization Finished
['feature_face_caption']
/home/arturo/relative_test3_NCEV.txt
/home/arturo/test3
1
500
ann_path:  /home/arturo/relative_test3_NCEV.txt
video number:834
Accuracy: 0.8932853717026379
Precision: 0.8918073472309769
Recall: 0.8932853717026379
F1 Score: 0.891229884725899
[[172   1   6   2   0   2]
 [  0 154  10   3   1   1]
 [  7  11 135   5   1   7]
 [  1   4   1 248   0   3]
 [  1   0   1   3   5   4]
 [  4   1   3   6   0  31]]
```

**Comment by ZebangCheng** [comment]
Thank you for your detailed feedback. After training Stage 1 with `MER2023_pseudo_label.txt`, did you evaluate the model performance at that stage? In our experiments, we evaluated multiple saved checkpoints during Stage 1 and selected the best-performing one to continue with Stage 2 training.

The discrepancy in your results might be due to differences in the selected Stage 1 checkpoint. If possible, could you share the evaluation results (accuracy/F1, etc.) for the various checkpoints saved during Stage 1? That would help us better understand where the difference may come from.

**Comment by ZebangCheng** [comment]
Thank you for providing the log information. I reviewed it and noticed that your Stage 1 performance is around 86%, which is slightly lower than ours — we achieved approximately 88%, and the best checkpoint reached around 89%. Based on that best-performing checkpoint, we proceeded with Stage 2 fine-tuning and were able to reach 90%.

One possible reason for the difference could be the batch size during training. In our setup, we used 4 GPUs with a batch size of 1 per GPU. Could you let us know what training setup you used on your side?

**Comment by ZebangCheng** [comment]
Looking forward to seeing your updated results. If your hardware allows, I would recommend trying the training with 4 GPUs to match our setup more closely.

Additionally, for optimal performance, you may consider commenting out the other prompts in the `emotion_instruction_pool` and using only a single prompt during both training and testing. However, please note that this could lead to overfitting and may reduce the model’s generalization ability.
